### DISCUSSION TITLE
How to encrypt cache?

### DISCUSSION THREAD

**Discussion by waltermvp** [TanStack/query#3562] [Q&A]
Let's say I need to encrypt data that is cached due to PHI regulations. How would I go about encrypting all the data that is saved to disk?

**Comment by TkDodo** [comment]
the cache is in memory only - unless you use the `persistQueryClientPlugin` - in which case you can write your own persister that takes care of encryption.

**Reply by waltermvp** [reply]
Thanks for the quick reply, things are making sense now :)

**Reply by MrT3313** [reply]
Link: https://tanstack.com/query/v4/docs/plugins/persistQueryClient

**Reply by mroswald** [reply]
Under normal conditions, this memory can not be accessed, but there are way to dump it when devices get out of hand for example. Just reading a value from an api and show it on a screen usually leads to a cleanup of memory when the screen is closed or navigated away. In the case of react-query caching it will be available until the app is force-fully closed or the device got restarted. Therefore, having a way to encrypt the cache would be helpful. Is there a way?
